src/common/ImageAvatar.js

```diff
@@ -31,14 +44,17 @@ export default class ImageAvatar extends PureComponent<Props> {
     const borderRadius =
       shape === 'rounded' ? size / 8 : shape === 'circle' ? size / 2 : shape === 'square' ? 0 : 0;
 
+    const imageSource = isLoading ? userDefaultAvatar : { uri: avatarUrl };
+
     return (
       <View>
         <Touchable onPress={onPress} style={touchableStyle}>
           <ImageBackground
             style={touchableStyle}
-            source={{ uri: avatarUrl }}
+            source={imageSource}
             resizeMode="cover"
             borderRadius={borderRadius}
+            onLoad={(e) => { this.setState({ isLoading: false }); }}
```

Comment:
We always do class-level functions for these events.
As in `onLoad={this.handleLoad}`
and then the function itself is
`handleLoad = (e) => { this.setState({ isLoading: false }); }`

If we do it inline like you have done it, it will recreate a new function for every component.